src/servers/src/http.rs

```diff
@@ -358,19 +361,50 @@ impl HttpServer {
         grpc_handler: ServerGrpcQueryHandlerRef,
         options: HttpOptions,
     ) -> Self {
+        let mut server = Self::new_empty(options);
+        server.set_sql_handler(sql_handler);
+        server.set_grpc_handler(grpc_handler);
+        server.set_metrics_handler(MetricsHandler);
+        server
+    }
+
+    pub fn new_empty(options: HttpOptions) -> Self {
```

Comment:
Prefer to use builder pattern, since we have too many fields in struct HttpServer.

Reply:
Add a builder(`HttpServerBuilder`) to build `HttpServer`